resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientUtils.scala

```diff
@@ -156,11 +162,8 @@ private[spark] object KubernetesClientUtils extends Logging {
     }
     val confFiles: Seq[File] = {
       val dir = new File(confDir)
-      if (dir.isDirectory) {
-        dir.listFiles.filter(x => fileFilter(x)).toSeq
-      } else {
-        Nil
-      }
+      assert(dir.isDirectory, "Spark conf should be a directory.")
```

Comment:
nit: move this assertion up so it's caught early (at least around line 135). I also think it should be `IllegalArgumentException` instead.

Reply:
In terms of execution, both have similar effect. Still if others also feel - that this should be moved, I can do so.